### DISCUSSION TITLE
Stream Body on Serverless closes connection early

### DISCUSSION THREAD

**Discussion by jregistr** [tokio-rs/axum#2041] [Q&A]
Hi everyone. I'm testing writing an endpoint that will send chunks of data with a StreamBody
```rust
let stream = stream.map(|item| ResponseChunk(item));
let body = StreamBody::new(stream);
Ok(body)
```
I used fetch to test my axum server running on localhost and I successfully get all the chunks until it is done.
```js
 const reader = response.body.getReader();
  while (true) {
    const { value, done } = await reader.read();
    if (done) break;
    let text = new TextDecoder().decode(value);
    console.log("text", text);
  }
```

But when I deployed my container to Google Cloud Run or Fly.io, it looks like the connection is being closed early. On cloud run I get this message:
```
{
resource: {2}
severity: "WARNING"
textPayload: "Truncated response body. Usually implies that the request timed out or the application exited before the response was finished.
}
```

I'm unsure if I'm misconfiguring axum or these platforms so I thought I'd ask here.

**Comment by manku211** [comment]
pls  suggest anyone

**Comment by mladedav** [comment]
Having a reproducer (on fly.io preferably) would help in diagnosing the issue.
